clients/mediaconvert_test.go

```diff
@@ -0,0 +1,298 @@
+			createJobCalls++
+			return &mediaconvert.CreateJobOutput{Job: &mediaconvert.Job{Id: aws.String("10")}}, nil
+		},
+		getJob: func(input *mediaconvert.GetJobInput) (*mediaconvert.GetJobOutput, error) {
+			getJobCalls++
+			switch getJobCalls {
+			case 1:
+				return &mediaconvert.GetJobOutput{Job: &mediaconvert.Job{
+					Status:             aws.String(mediaconvert.JobStatusProgressing),
+					JobPercentComplete: aws.Int64(50),
+				}}, nil
+			case 2:
+				require.NoError(os.WriteFile(transfOutputFile, []byte(dummyHlsPlaylist), 0777))
+				require.NoError(os.WriteFile(path.Join(transfOutputFile, "../1.ts"), []byte(exampleFileContents), 0777))
+
+				return &mediaconvert.GetJobOutput{Job: &mediaconvert.Job{
+					Status: aws.String(mediaconvert.JobStatusComplete),
+				}}, nil
+			default:
+				require.Fail("unexpected call")
+				return nil, errors.New("unreachable")
+			}
+		},
+	}
+	mc, inputFile, transferDir, cleanup := setupTestMediaConvert(t, awsStub)
+	defer cleanup()
+
+	outFile := path.Join(transferDir, "../out/index.m3u8")
+	defer os.RemoveAll(path.Dir(outFile))
+	transfOutputFile = path.Join(transferDir, "output", outFile)
+	require.NoError(os.MkdirAll(path.Dir(transfOutputFile), 0777))
+
+	err := mc.Transcode(context.Background(), TranscodeJobArgs{
+		InputFile:                mustParseURL("file://" + inputFile.Name()),
+		HLSOutputFile:            mustParseURL("file:/" + outFile),
+		CollectSourceSize:        func(size int64) {},
+		ReportProgress:           func(progress float64) {},
+		CollectTranscodedSegment: func() {},
+	})
+	require.NoError(err)
+	require.Equal(1, createJobCalls)
+	require.Equal(2, getJobCalls)
+
+	// Check that the output files were copied to the osTransferBucketURL folder
+	content, err := os.ReadFile(outFile)
+	require.NoError(err)
+	require.Equal(dummyHlsPlaylist, string(content))
+
+	content, err = os.ReadFile(path.Join(outFile, "../1.ts"))
+	require.NoError(err)
+	require.Equal(exampleFileContents, string(content))
+}
+
+func setupTestMediaConvert(t *testing.T, awsStub AWSMediaConvertClient) (mc *MediaConvert, inputFile *os.File, transferDir string, cleanup func()) {
+	oldMaxRetryInterval, oldRetries, oldPollDelay := maxRetryInterval, config.DownloadOSURLRetries, pollDelay
+	maxRetryInterval, config.DownloadOSURLRetries, pollDelay = 1*time.Millisecond, 1, 1*time.Millisecond
+
+	var err error
+	inputFile, err = os.CreateTemp(os.TempDir(), "user-input-*")
+	require.NoError(t, err)
+	_, err = inputFile.WriteString(exampleFileContents)
+	require.NoError(t, err)
+	require.NoError(t, inputFile.Close())
+
+	// use the random file name as the dir name for the transfer file
+	transferDir = path.Join(inputFile.Name()+"-dir", "transfer")
+	require.NoError(t, os.MkdirAll(transferDir, 0777))
+
+	cleanup = func() {
+		maxRetryInterval, config.DownloadOSURLRetries, pollDelay = oldMaxRetryInterval, oldRetries, oldPollDelay
+		require.NoError(t, os.Remove(inputFile.Name()))
```

Comment:
If this fails, it'll not run the next step that removes the dir - this might actually be a case where we want `assert` instead of `require`

Reply:
Doesn't seem like we're using `asset` in `catalyst-api` so I just set the err on a variable to check in the end here.